cucumber-core/README.md

```diff
@@ -79,6 +80,22 @@ They are  respectively responsible for discovering glue classes, registering
 step definitions, and creating instances of said glue classes. Backend and
 object factory implementations are discovered via SPI.
 
+## Order ##
```

Comment:
Consider structuring this as 

- Explain what we are talking about (i.e. we can change the order of pickles).
- Explain each build in ordering individually and in detail 
- Explain that custom orders can be implemented by implementing the `PickleFactory` interface.

Move all other details to the Javadoc. People who are going to implement the interface will see it there.